### DISCUSSION TITLE
获取语言版本直接报错

### DISCUSSION THREAD

**Discussion by singgl** [umijs/umi#11471] [General]
import { getLocale } from 'umi';
![image](https://github.com/umijs/umi/assets/26128398/a1044db5-197e-4174-b51a-e76b3bee2c7b)

**Comment by fz6m** [comment]
需要给一个最小复现。

**Comment by singgl** [comment]
就是cli下直接在app.tsx中引入import { getLocale } from 'umi';
在直接console.log('语言类型', getLocale());就报错
![image](https://github.com/umijs/umi/assets/26128398/d0f2fba5-b84d-4f85-9c3f-82623a364d60)

**Reply by fz6m** [reply]
看一下你 `app.tsx` 的调用方式，给一个最小复现仓库吧。

**Comment by singgl** [comment]
![image](https://github.com/umijs/umi/assets/26128398/f7e95b6b-8768-4f18-973c-353d64d5a014)
调用就是如此,执行ci后,啥也没加

**Reply by fz6m** [reply]
这个 `app.tsx` 最外层是全局，没法执行这个函数，因为这个时机还没启动应用，i18n 都没初始化好，你要在启动应用之后再调用，比如实际使用的时候。

**Comment by singgl** [comment]
![image](https://github.com/umijs/umi/assets/26128398/2db90e3f-d280-4910-83e5-47fc1e7f6017)
![image](https://github.com/umijs/umi/assets/26128398/420879e3-fd68-4711-8ab6-c85b5b121eba)
在我umi3项目里是可以拿到的
